scripts/init.groovy.d/ab_configure_github_oauth.groovy

```diff
@@ -0,0 +1,115 @@
+// output init hook stage
+println '### JENKINS AUTOCONFIG: GITHUB OAUTH ###'
+
+/**
+  * Check if Jenkins should not configure itself and skip.
+  */
+
+Boolean skipConfiguringJenkins() {
+    !(System.env?.ENVIRONMENT in ['dev', 'staging']) ||
+        (new File("${Jenkins.instance.rootDir}/autoConfigComplete").exists())
+}
+if(skipConfiguringJenkins()) {
+    println 'Skipping Jenkins auto-configuration'
+    return
+}
+
+
+/**
+  * From this point onward Jenkins should configure itself.
+  */
+import jenkins.*
+import jenkins.model.*
+import hudson.*
+import hudson.model.*
+
+import org.yaml.snakeyaml.LoaderOptions
+import org.yaml.snakeyaml.Yaml
+import org.yaml.snakeyaml.constructor.SafeConstructor
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse
+
+import hudson.security.SecurityRealm
+import org.jenkinsci.plugins.GithubSecurityRealm
+import net.sf.json.JSONObject
+
+Map getSecret(Map options) {
+    String secretName = options.secretId
+    Region region = Region.of(options.region)
+    SecretsManagerClient client = SecretsManagerClient.builder()
+        .region(region)
+        .build()
+    GetSecretValueRequest getSecretValueRequest = GetSecretValueRequest.builder()
+        .secretId(secretName)
+        .build()
+    GetSecretValueResponse getSecretValueResponse = client.getSecretValue(getSecretValueRequest)
+    String secret = getSecretValueResponse.secretString()
+    LoaderOptions loaderOptions = new LoaderOptions()
+    def yaml = new Yaml(new SafeConstructor(loaderOptions))
+    yaml.load(secret)
+}
+String ENVIRONMENT = System.env?.ENVIRONMENT
+String HALF = System.env?.HALF
+
+// list of github realm maps
+github_realm = getSecret(region: 'us-east-1', secretId: "jenkins-ng/oauth/github")."${ENVIRONMENT}"."${HALF}".collectEntries()
+
+/*
+   Copyright (c) 2015-2020 Sam Gleske - https://github.com/samrocketman/jenkins-bootstrap-jervis
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+   */
```

Comment:
Potential suggestion (not blocking): should we move this copyright header to the top of the file instead?

Reply:
i dont even know if it should be included since its a part of modified code. Might be a question to bring up to Sam.

Reply:
for now i've moved to the top of the file